app/means_test/api.py

```diff
@@ -64,41 +40,50 @@ def get_means_test_payload(eligibility_data: EligibilityData) -> dict:
     is_partner_employed = about.get("is_partner_employed", None)
     is_partner_self_employed = about.get("is_partner_self_employed", None)
 
-    income_data: dict[str, dict] = IncomeForm(**income_form).get_payload(
+    benefits_data = BenefitsForm.get_payload(eligibility_data.forms.get("benefits", {}))
+    additional_benefits_data = AdditionalBenefitsForm.get_payload(
+        eligibility_data.forms.get("additional-benefits", {})
+    )
+    income_data = IncomeForm(**eligibility_data.forms.get("income", {})).get_payload(
```

Comment:
What do you think of passing eligiblity_data object to get_payload and let it extract the fields from it on it's own?